File: app/src/main/java/be/digitalia/fosdem/db/converters/NullableDateTypeConverters.java
package be.digitalia.fosdem.db.converters;

import java.util.Date;

import androidx.room.TypeConverter;

public class NullableDateTypeConverters {
	@TypeConverter
	public static Date toDate(Long value) {
		return value == null ? null : new Date(value);
	}

	@TypeConverter
	public static Long fromDate(Date value) {
		return (value == null) ? null : value.getTime();
	}
}

File: app/src/main/java/be/digitalia/fosdem/db/converters/NullableDateTypeConverters.kt
package be.digitalia.fosdem.db.converters

import androidx.room.TypeConverter
import java.util.Date

object NullableDateTypeConverters {
    @JvmStatic
    @TypeConverter
    fun toDate(value: Long?): Date? = value?.let { Date(it) }

    @JvmStatic
    @TypeConverter
    fun fromDate(value: Date?): Long? = value?.time
}